common/net-message.cpp

```diff
@@ -0,0 +1,233 @@
+
+#include <spdlog/spdlog.h>
+
+#include "net-message.hpp"
+
+namespace tec {
+namespace networking {
+
+void MessageOut::FromBuffer(const void* body, size_t length) {
+	size_t remain = length;
+	size_t offset = 0;
+	const char* body_ptr = reinterpret_cast<const char*>(body);
+	while (remain > 0) {
+		void* buffer;
+		int buffer_size;
+		if (!Next(&buffer, &buffer_size)) {
+			spdlog::get("console_log")->warn("MessageStream::FromString failed to request buffer");
+			return;
+		}
+		if (!buffer_size) {
```

Comment:
If buffer size where to be -1 somehow this would return false.

Reply:
We implement `Next` (see it's definition below).
As part of the [requirements for a ZeroCopyStream](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.io.zero_copy_stream#ZeroCopyOutputStream.Next.details), the `size` parameter will be either unchanged with `Next` returning false or set to the number of bytes returned. Returning a negative number of bytes would violate that contract.
In this implementation, the size is sourced from an unsigned count of bytes.
If the buffer size is less than zero, then there are other problems that would need fixing.